src/runtime_src/xdp/profile/plugin/ocl/xocl_plugin.cpp

```diff
@@ -275,6 +281,31 @@ namespace xdp {
     }
   }
 
+  void XoclPlugin::getPlramSizeDevices()
+  {
+    for (auto device : mPlatformHandle->get_device_range()) {
+      if (!device->is_active())
+        continue;
+      auto name = device->get_unique_name();
+      auto sz = xdp::xoclp::platform::device::getPlramSizeBytes(device);
+      if (sz)
+        mDevicePlramSizeMap[name] = sz;
+    }
+  }
+
+  void XoclPlugin::getMemBitWidthDevices()
+  {
+    for (auto device : mPlatformHandle->get_device_range()) {
+      if (!device->is_active())
+        continue;
+      // TODO: figure out how to get this from platform
+      auto name = device->get_unique_name();
+      mDeviceMemTypeBitWidthMap[name + "|HBM"] = 256;
+      mDeviceMemTypeBitWidthMap[name + "|DDR"] = 512;
+      mDeviceMemTypeBitWidthMap[name + "|PLRAM"] = 512;
```

Comment:
Potentially in a future checkin, please add support for SoC platforms.

Reply:
@IshitaGhosh  is adding a way to distinguish SOC from PCIE. Will add this after her checkin